apps/sms/views/shared/js/navigation.js

```diff
@@ -491,28 +491,21 @@ function executeNavigationStep(stepName) {
 }
 
 /**
- * Attach a handler to run the `afterEnter` step after the page is loaded.
- * It uses either the custom event `navigation-transition-end` or the usual
- * `load` event.
- * @returns {Promise} Resolved once afterEnter is run.
+ * Returns promise that is resolved either once Navigation object is marked as
+ * ready (non split views) or when custom `navigation-transition-end` event is
+ * fired (split views).
+ * @returns {Promise.<void>}
  */
-function attachAfterEnterHandler() {
-  if (document.readyState === 'complete') {
-    return executeNavigationStep('afterEnter').catch(catchStepError);
-  }
-
-  var defer = Utils.Promise.defer();
+function waitForThePanelToBeReady() {
+  var readyPromise = readyDefer && readyDefer.promise || Promise.resolve();
```

Comment:
```readyDefer && readyDefer.promise``` - I see there is a theoretical chance that ```readyDefer``` is nullified before we call this method.

``` || Promise.resolve()``` just for the consistency of returned result.